### ISSUE TITLE
MATOMO_SMTP_PROTOCOL of `none` is ignored by PHP.

### ISSUE THREAD

**Issue by nepella** [bitnami/bitnami-docker-matomo#104]
### Name and version
bitnami/matomo:4.7.1-debian-10-r14

### The set-up
1. Set the service's environment variable `MATOMO_SMTP_PROTOCOL` to `none` to disable opportunistic STARTTLS.
2. Start the Matomo service with an empty volume to force initialization.
3. Watch the `mail.encryption` value in `/bitnami/matomo/config/config.ini.php`:
    ```sh
    while :; do
        docker exec -i $(docker ps -qfname=matomo_app) \
            grep -A 10 '^\[mail\]' /bitnami/matomo/config/config.ini.php
        sleep 1
    done
    ```

### The problem
After the entrypoint reports "Configuring SMTP credentials," the correct setting briefly appears:

```
[mail]
transport=smtp
port=25
host=myhost
username=
password=
type=
encryption=none
```

Then, shortly after Apache starts, the setting disappears:

```
[mail]
transport = "smtp"
port = 25
host = "myhost"

[PluginsInstalled]
PluginsInstalled[] = "CustomDimensions"
PluginsInstalled[] = "CoreVue"
```

### The cause
The trouble is almost certainly that PHP's [`parse_ini_file`](https://www.php.net/manual/en/function.parse-ini-file.php) interprets `none` as null (unless it's called with `scanner_mode` set to `INI_SCANNER_RAW`). The correct setting appears after `libmatomo.sh` sets the encryption setting and disappears after Matomo parses and rewrites the file.

### Remedies
Adding quotes to the value passed to `ini-file` causes it to write a value which PHP parses correctly:

```sh
ini-file set -s "mail" -k "encryption" -v "'$MATOMO_SMTP_PROTOCOL'" "$MATOMO_CONF_FILE"
```

But it may be better to alter `ini-file` itself to handle `none` as a value. (It knows to quote strings containing `#`.)

**Comment by carrodher** [comment]
Please, [note the Bitnami team is not actively monitoring GitHub Issues for this project](https://github.com/bitnami/bitnami-docker-matomo#community-supported-solution). This issue will stay open for 20 days to allow the community to contribute, after 15 days without activity the issue will be marked as stale being closed after 5 days.

If you would like to contribute to this container image by fixing a bug or adding a new feature feel free to create a PR, the Bitnami team will be happy to review it and provide feedback.

**Comment by nepella** [comment]
@carrodher: Two years ago, you solved #45. This issue is similar. Please read what I wrote under "additional information" above: I don't know whether adding a kludge for this instance or trying to correct the problem in `ini-file` is preferred.

In case my proposed kludge isn't obvious, here is the working diff:

```diff
--- a/4/debian-10/rootfs/opt/bitnami/scripts/libmatomo.sh
+++ b/4/debian-10/rootfs/opt/bitnami/scripts/libmatomo.sh
@@ -185,3 +185,3 @@
             ini-file set -s "mail" -k "type" -v "$MATOMO_SMTP_AUTH" "$MATOMO_CONF_FILE"
-            ini-file set -s "mail" -k "encryption" -v "$MATOMO_SMTP_PROTOCOL" "$MATOMO_CONF_FILE"
+            ini-file set -s "mail" -k "encryption" -v "'$MATOMO_SMTP_PROTOCOL'" "$MATOMO_CONF_FILE"
         fi
```
